File: SmartPlanner/Domain/Tasks/TaskValidator.swift
//
//  TaskValidator.swift
//  SmartPlanner
//

import Foundation

protocol TaskValidating {
    func isValidTitle(_ title: String) -> Bool
    func sanitize(_ text: String?) -> String?
}

struct TaskValidator: TaskValidating {

    enum Constants {
        static let titleMaxLength = 80
        static let titleMinLength = 1
    }

    func isValidTitle(_ title: String) -> Bool {
        let trimmed = title.trimmingCharacters(in: .whitespacesAndNewlines)
        return trimmed.count >= Constants.titleMinLength
            && trimmed.count <= Constants.titleMaxLength
    }

    func sanitize(_ text: String?) -> String? {
        guard let text else { return nil }
        let trimmed = text.trimmingCharacters(in: .whitespacesAndNewlines)
        return trimmed.isEmpty ? nil : trimmed
    }
}

File: Android/app/src/main/java/com/example/smartplannercompose/domain/tasks/TaskValidator.kt
package com.example.smartplannercompose.domain.tasks

interface TaskValidator {
    fun isValidTitle(title: String): Boolean
    fun sanitize(text: String?): String?

    companion object {
        const val TITLE_MIN_LENGTH = 1
        const val TITLE_MAX_LENGTH = 80
    }
}

class DefaultTaskValidator : TaskValidator {
    override fun isValidTitle(title: String): Boolean {
        val trimmed = title.trim()
        return trimmed.length in TaskValidator.TITLE_MIN_LENGTH..TaskValidator.TITLE_MAX_LENGTH
    }

    override fun sanitize(text: String?): String? {
        if (text == null) return null
        val trimmed = text.trim()
        return trimmed.ifEmpty { null }
    }
}
